.circleci/config.yml

```diff
@@ -159,6 +160,28 @@ jobs:
       - run: TELEPRESENCE_VERSION=$CIRCLE_TAG make push-executable
       - run: |
           [[ $CIRCLE_TAG == *-* ]] || TELEPRESENCE_VERSION=$CIRCLE_TAG make promote-to-stable
+          [[ $CIRCLE_TAG == *-nightly-* ]] && TELEPRESENCE_VERSION=$CIRCLE_TAG make promote-nightly
+
+  "tag-nightly":
+    executor: vm-linux
+    steps:
+    - checkout
+    - install-go
+    - run:
+        name: "Tag nightly"
+        command: |
+          existingTags=$(git tag --points-at HEAD)
+          if [[ -n $existingTags ]]; then
+              echo "Tag for the HEAD of release/v2 already exists:"
+              echo $existingTags
+              exit 0
+          else
+              lastTag=$(git describe --tags --match="v*" --abbrev=0
+              time=$(date +"%s")
+              newTag="$lastTag-nightly-$time"
```

Comment:
1. Isn't this going to recurse?  On the first night it works correct, and tags `v2.3.2-nightly-1624775772`.  Then on the second day it tags `v2.3.2-nightly-1624775772-nightly-1624862232`, and so on?
2. I'd prefer to have yyyy.mm.dd rather than seconds since the epoch.  Make it easier for people to talk about which one they have.
3. That tag format isn't semver.  `vX.Y.Z-P` indicates something *before* X.Y.Z, but we're actually tagging something *after* it.

Reply:
I'm thinking
```console
$ git describe --tags --match 'v*' --exclude='*-*'|sed "s/-/-nightly.$(date +%Y.%m.%d)-/"
v2.3.2-nightly.2021.06.27-19-ge9b3c27f
```
but with the Z bumped.  Probably reuse `genversion.go`?

Reply:
1. I think I misunderstood how --abbrev=0 worked, so I believe you are right.
2. Noted + made that change.
3. Ah at last job we used vX.Y.Z-P to be something after vX.Y.Z and before vX.Y.Z+1, so I guess we were using semver wrong, thanks for letting me know.

I originally didn't use `genversion.go` since the format of the tags was different, but it probably makes sense to include all our version-related things there, so have updated it.